packages/python/accern_data/accern_data.py

```diff
@@ -303,45 +365,65 @@ def validate_filters(filters: Optional[FiltersType]) -> FiltersType:
         if filters is None:
             return {}
         for key in filters.keys():
-            if key not in FilterField:
+            if key not in FILTER_FIELD:
                 raise ValueError(
-                    f"{key} is not a valid filed."
-                    f"Possible fileds: {FilterField}")
+                    f"{key} is not a valid field."
+                    f"Possible fields: {FILTER_FIELD}")
+            if key == "primary_signal":
+                val = filters["primary_signal"]
+                if isinstance(val, bool):
+                    filters["primary_signal"] = f"{val}".lower()
         return filters
 
     def set_filters(self, filters: FiltersType) -> None:
         self._filters = self.validate_filters(filters)
```

Comment:
set_filters and set_raw_filters do the same thing. only change is set_filters check whether all fields are known.

Reply:
yes